### ISSUE TITLE
Lint/UnneededCopDisableDirective removes valid comments

### ISSUE THREAD

**Issue by ghiculescu** [rubocop-hq/rubocop#7020]
`Metrics/LineLength` config:

```yaml
Metrics/LineLength:
  Max: 140
  IgnoreCopDirectives: true
```

Code:

```ruby
"long long long long long long long long long long long long long long long long long long long long long long long line" # a long line

"long long long long long long long long long long long long long long long long long long long long long long long line" # a long line # rubocop:disable Metrics/ClassLength
```

The cop that's disabled in the second string is irrelevant. The point is that we are disabling a cop that *isn't* `Metrics/LineLength`

Run rubocop on this file:

```
test.rb:3:123: W: Lint/UnneededCopDisableDirective: Unnecessary disabling of Metrics/ClassLength.
"long long long long long long long long long long long long long long long long long long long long long long long line" # a long line # rubocop:disable Metrics/ClassLength
                                                                                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Cool. Let's autocorrect the issue, `bundle exec rubocop -a test.rb`. Output:

```
test.rb:3:123: W: [Corrected] Lint/UnneededCopDisableDirective: Unnecessary disabling of Metrics/ClassLength.
"long long long long long long long long long long long long long long long long long long long long long long long line" # a long line # rubocop:disable Metrics/ClassLength
                                                                                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

And the file contents are now:

```ruby
"long long long long long long long long long long long long long long long long long long long long long long long line" # a long line

"long long long long long long long long long long long long long long long long long long long long long long long line"
```

Basically, the "a long line" part of the 2nd comment has been removed.

## Expected behavior

`# rubocop:disable Metrics/ClassLength` is removed, but `# a long line` is not.

## Actual behavior

See above.

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ bundle exec rubocop -V
0.68.1 (using Parser 2.6.3.0, running on ruby 2.5.3 x86_64-darwin15)
```

**Comment by mikegee** [comment]
I just encountered this bug too. I had code like:

```ruby
# my long and important comment # rubocop:disable Metrics/LineLength
```

Running `rubocop --safe-auto-correct` deletes that whole line.
